packages/wrangler/src/pages/dev.ts

```diff
@@ -229,99 +236,86 @@ export function Options(yargs: CommonYargsArgv) {
 	}
 
-	if (experimentalLocal) {
+	if (args.experimentalLocal) {
 		logger.warn(
 			"--experimental-local is no longer required and will be removed in a future version.\n`wrangler pages dev` now uses the local Cloudflare Workers runtime by default."
 		);
 	}
 
-	if (config) {
-		throw new FatalError("Pages does not support wrangler.toml", 1);
+	if (args.config) {
+		throw new FatalError(
+			"Pages does not support custom paths for the `wrangler.toml` configuration file",
+			1
+		);
 	}
 
-	if (singleWorkerScriptPath !== undefined) {
+	if (args.scriptPath !== undefined) {
 		logger.warn(
 			`\`--script-path\` is deprecated and will be removed in a future version of Wrangler.\nThe Worker script should be named \`_worker.js\` and located in the build output directory of your project (specified with \`wrangler pages dev <directory>\`).`
 		);
 	}
 
-	singleWorkerScriptPath ??= DEFAULT_SCRIPT_PATH;
+	const config = readConfig(undefined, args);
+	const {
+		directory: resolvedDirectory,
+		compatibilityDate,
+		compatibilityFlags,
+		ip,
+		port,
+		inspectorPort,
+		localProtocol,
+	} = resolvePagesDevServerSettings(config, args);
 
+	const [_pages, _dev, ...remaining] = args._;
 	const command = remaining;
-
 	let proxyPort: number | undefined;
+	let directory = resolvedDirectory;
 
 	if (directory !== undefined && command.length > 0) {
 		throw new FatalError(
 			"Specify either a directory OR a proxy command, not both.",
 			1
 		);
 	} else if (directory === undefined) {
-		logger.warn(
-			`Specifying a \`-- <command>\` or \`--proxy\` is deprecated and will be removed in a future version of Wrangler.\nBuild your application to a directory and run the \`wrangler pages dev <directory>\` instead.\nThis results in a more faithful emulation of production behavior.`
-		);
-
 		proxyPort = await spawnProxyProcess({
-			port: requestedProxyPort,
+			port: args.proxy,
 			command,
 		});
 		if (proxyPort === undefined) return undefined;
 	} else {
 		directory = resolve(directory);
 	}
 
-	if (!compatibilityDate) {
-		const currentDate = new Date().toISOString().substring(0, 10);
-		logger.warn(
-			`No compatibility_date was specified. Using today's date: ${currentDate}.\n` +
-				"Pass it in your terminal:\n" +
-				"```\n" +
-				`--compatibility-date=${currentDate}\n` +
-				"```\n" +
-				"See https://developers.cloudflare.com/workers/platform/compatibility-dates/ for more information."
-		);
-		compatibilityDate = currentDate;
+	const {
+		vars,
+		kv_namespaces,
+		do_bindings,
+		d1_databases,
+		r2_buckets,
+		services,
+		ai,
+	} = getBindings(args);
+
+	if (services && services.find(({ environment }) => !!environment)) {
+		// We haven't yet properly defined how environments of service bindings should
+		// work, so if the user is using an environment for any of their service
+		// bindings we warn them that they are experimental
+		logger.warn("Support for service binding environments is experimental.");
 	}
```

Comment:
I think it would be nice to move this to the `getBindings()` function.

Reply:
do you mean the one in this file?

Reply:
Yep

Reply:
very good callout! fixed!